service/discovery/azure/compute.go

```diff
@@ -724,9 +618,179 @@ func (d *azureComputeDiscovery) keyURL(diskEncryptionSetID string) (string, *arm
 	return util.Deref(keyURL), &kv.DiskEncryptionSet, nil
 }
 
+// getSiteProperties returns properties for WebApp und functions
+func (d *azureComputeDiscovery) getSiteProperties(site *armappservice.Site) (ni []voc.ResourceID, publicAccess bool, resourceLogging *voc.ResourceLogging) {
+
+	ni = []voc.ResourceID{}
+	resourceLogging = &voc.ResourceLogging{}
+
+	// If a mandatory field is empty, the whole function is empty
+	if site == nil {
+		return
+	}
+
+	// Get virtual network subnet ID
+	if site.Properties.VirtualNetworkSubnetID != nil {
```

Comment:
Properties shouldn't be ever nil. Otherwise, it could panic here. But I guess, Properties is a mandatory field of site

Reply:
Couldn't we use/build a util.deref function that takes into account the different levels and check if they are nil? I guess not or it is very hard

Reply:
I don't know exactly how to handle this either. And I also thought about a extended Deref-function. Maybe we could convert your comment to an issue.